server/rest/middleware/usersignup_test.go

```diff
@@ -0,0 +1,205 @@
+package middleware_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+
+	ccontext "github.com/konflux-workspaces/workspaces/server/core/context"
+	"github.com/konflux-workspaces/workspaces/server/rest/middleware"
+	"github.com/konflux-workspaces/workspaces/server/rest/middleware/mocks"
+)
+
+const (
+	endpointWhatever = "/whatever"
+
+	methodGet = "GET"
+
+	testUserSub = "test-user-sub"
+)
+
+var _ = Describe("Usersignup", func() {
+
+	var (
+		ctx context.Context
+
+		// mocks
+		c *mocks.MockFakeCRCache
+		h *mocks.MockFakeHTTPHandler
+
+		// http
+		w *httptest.ResponseRecorder
+		r *http.Request
+
+		// middleware
+		m *middleware.UserSignupMiddleware
+	)
+
+	BeforeEach(func() {
+		ctx = context.TODO()
+
+		// mocks
+		ctrl := gomock.NewController(GinkgoT())
+		c = mocks.NewMockFakeCRCache(ctrl)
+		h = mocks.NewMockFakeHTTPHandler(ctrl)
+
+		// http
+		w = httptest.NewRecorder()
+		b := bytes.NewBuffer([]byte{})
+		lr, err := http.NewRequestWithContext(ctx, methodGet, endpointWhatever, b)
+		Expect(err).NotTo(HaveOccurred())
+		r = lr
+
+		// middleware
+		m = middleware.NewUserSignupMiddleware(h, c, true)
+	})
+
+	When("sub HTTP header is not present", func() {
+		It("invokes next handler", func() {
+			// set expectations
+			c.EXPECT().List(gomock.Any(), gomock.Any()).Times(0)
+			h.EXPECT().ServeHTTP(gomock.Any(), gomock.Any()).Times(1)
+
+			// when
+			m.ServeHTTP(w, r)
+		})
```

Comment:
This test doesn't assert anything?

Reply:
fixed in b9dd1b2636210e1bc7f5f1437359aa60de68c9f2